contracts/OperatorRegistry.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: UNLICENSED
+// ⣿⣿⣿⣿⣿⠀⠀⣰⣿⣿⣿⣷⡀⠀⠀⣶⣶⣶⣦⡀⠀⠀⠀⣶⣶⡄⠀⠀⣶⣶⡆⠀⠀⣶⣶⠀⠀⠀⠀⢰⣶⣶⣶⣶⢀⠀⠀⣤⣶⣶⣦⡀⠀⠀⠀⣴⣶⣶⣦⠀
+// ⣿⣿⠀⠀⠀⠀⠀⣿⣿⠀⢸⣿⡇⠀⠀⣿⣿⠀⢻⣿⠀⠀⠀⣿⣿⣿⠀⢸⣿⣿⡇⠀⠀⣿⣿⠀⠀⠀⠀⢸⣿⡇⠀⠀⠀⠀⢸⣿⡇⠀⣿⣿⠀⠀⣾⣿⠁⠈⣿⡇
+// ⣿⣿⠀⠀⠀⠀⠀⣿⣿⠀⢸⣿⡇⠀⠀⣿⣿⠀⣸⣿⠀⠀⠀⣿⣿⣿⡀⣿⡟⣿⡇⠀⠀⣿⣿⠀⠀⠀⠀⢸⣿⡇⠀⠀⠀⠀⠀⣿⣿⡀⠀⠀⠀⠀⠘⣿⣷⠀⠀⠀
+// ⣿⣿⠿⠿⠀⠀⠀⣿⣿⠀⢸⣿⡇⠀⠀⣿⣿⣿⣿⡟⠀⠀⠀⣿⣿⣿⣷⣿⠀⣿⡇⠀⠀⣿⣿⠀⠀⠀⠀⢸⣿⡿⠿⠀⠀⠀⠀⠀⢿⣿⣦⠀⠀⠀⠀⠈⣿⣿⡄⠀
+// ⣿⣿⠀⠀⠀⠀⠀⣿⣿⠀⢸⣿⡇⠀⠀⣿⣿⠈⣿⣷⠀⠀⠀⣿⣿⢸⣿⣿⠈⣿⡇⠀⠀⣿⣿⠀⠀⠀⠀⢸⣿⡇⠀⠀⠀⠀⢀⣀⠀⠙⣿⣧⠀⠀⣀⣀⠀⠻⣿⡆
+// ⣿⣿⠀⠀⠀⠀⠀⢿⣿⣤⣾⣿⠇⠀⠀⣿⣿⠀⣿⣿⠀⠀⠀⣿⣿⠀⣿⡇⠈⣿⡇⠀⠀⣿⣿⣤⣤⡄⠀⢸⣿⣧⣤⣤⡄⠀⢸⣿⣆⠀⣿⣿⠀⠀⣿⣿⡀⢀⣿⣿
+// ⠛⠛⠀⠀⠀⠀⠀⠈⠛⠿⠿⠛⠀⠀⠀⠛⠛⠀⠘⠛⠃⠀⠀⠛⠛⠀⠛⠀⠈⠛⠃⠀⠀⠛⠛⠛⠛⠃⠀⠘⠛⠛⠛⠛⠃⠀⠀⠙⠿⠿⠟⠁⠀⠀⠀⠛⠿⠿⠛⠀
+// https://formless.xyz/opportunities
+//
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @title Share Operator Registry contract.
+/// @notice A living canonical registry of all SHARE protocol operator
+/// EOAs (Externally Owned Accounts).
+/// @author john-paul@formless.xyz
+contract OperatorRegistry is Ownable {
+    // Using OpenZeppelin's EnumerableSet utility for managing addresses
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // Private set of operator addresses
+    EnumerableSet.AddressSet private _operatorAddresses;
+
+    /// @notice Initialize the contract with an initial set of operator addresses.
+    /// @param shareEOAOperators_ array of operator addresses to add to the registry initially
+    function initialize(address[] memory shareEOAOperators_) public onlyOwner {
+        for (uint256 i = 0; i < shareEOAOperators_.length; i++) {
+            _operatorAddresses.add(shareEOAOperators_[i]);
+        }
+    }
+
+    /// @notice Add a new verified operator to the registry.
+    /// @param shareOperatorAddress_ the address of the operator to add.
+    function addVerifiedOperator(
+        address shareOperatorAddress_
+    ) public onlyOwner {
+        _operatorAddresses.add(shareOperatorAddress_);
+    }
+
+    /// @notice Remove a verified operator from the registry.
+    /// @param shareOperatorAddress_ the address of the operator to remove.
+    function removeVerifiedOperator(
+        address shareOperatorAddress_
+    ) public onlyOwner {
+        _operatorAddresses.remove(shareOperatorAddress_);
+    }
+
+    /// @notice Provide funds to all registered operators. The funds sent with the transaction
+    /// will be evenly distributed amongst all registered operators.
+    function fundOperatorAddresses() public payable onlyOwner {
+        require(msg.value > 0, "SHARE033");
+        require(_operatorAddresses.length() > 0, "SHARE034");
+        uint256 fundingPerOperator = msg.value / _operatorAddresses.length();
```

Comment:
@un7c0rn  I agree, but where would the logic go and how would this look from a funding perspective(ie is it a script?)

Reply:
@DeusExM4ch1n4 No it can just be done manually and then the transfer can be invoked directly from the blockscanner or truffle. So for example the workflow would be 1. decide full funding amount 2. divide that by # of resource EOAs 3. pass both args to the funding function. This means we should have a `countOperatorAddresses`.